### ISSUE TITLE
Alien::Base::dynamic_libs does not find name for system install on OSX

### ISSUE THREAD

**Issue by shawnlaffan** [Perl5-Alien/Alien-Build#71]
If I run a system install of Alien::gdal on OSX, with a dependency on Alien::geos::af, then the system does not find the dynamic libs.  

Example result from a travis run is at Line 1088 https://travis-ci.org/shawnlaffan/perl-alien-gdal/jobs/454398792#L1088

This does not happen without Alien::geos::af, so it could be related to the GEOS library depending on an extra lib (libgeos_c) that is in the same location as libgeos, but which is not listed across all cflags/lib etc entries in the Alien::geos::af alien.json file.  

Alien::geos::af is at https://github.com/shawnlaffan/perl-alien-geos

There is also no .pc file for the GEOS library (if relevant).  

I have code in the Alien::gdal alienfile to append the geos libs to gdal.pc under a shared install, but it would be poor form to update a system gdal.pc.  

Any suggestions for tracking this down?

**Comment by plicease** [comment]
Does this work on other platforms say Linux?  Nothing stand out right now, but if I get a chance I can take a look late.

**Comment by shawnlaffan** [comment]
This is using homebrew, both on a travis instance and on a local machine.  I have not yet tried a share install on OSX but will give that a go today.  

It could also be how I'm handling the dynamic libs to append the GEOS libs to those for GDAL. 
https://github.com/shawnlaffan/perl-alien-gdal/blob/include_geos/lib/Alien/gdal.pm#L25

**Comment by shawnlaffan** [comment]
Looks like it's an Alien::geos::af issue when it runs a system install.  

Providing a dynamic_libs method in Alien:::geos::af seems to fix things.  For a system install it explicitly searches for the two libs that GEOS provides.  Otherwise it is uses the code in Alien::Base::dynamic_libs. 
https://github.com/shawnlaffan/perl-alien-geos/commit/92651d0b8f72a15fff55c5bed81c5961e46501ad

It would be useful if there were a generic way of handling cases like this, but I don't know if it is feasible.  GEOS does not provide a package config file that can be used - instead it has its own geos-config script.  I have code in the Alien::geos::af alienfile to process and append the various fields to the state.json file during the gather phase, but other aliens won't do this by default.

**Comment by plicease** [comment]
yeah interesting.  I want to leave this open for now and I will see if there is a good way to handle this in base class.  This might be a case where it makes sense to override as you've done here.
